core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerPool.scala

```diff
@@ -193,34 +238,27 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
       }
 
     // Container is free to take more work
-    case NeedWork(data: WarmedData) =>
+    case NeedWork(warmData: WarmedData) =>
       feed ! MessageFeed.Processed
-      if (data.activeActivationCount < data.action.limits.concurrency.maxConcurrent) {
+      val oldData = freePool.get(sender()).getOrElse(busyPool(sender()))
+      val newData = warmData.copy(activeActivationCount = oldData.activeActivationCount - 1)
+      if (newData.activeActivationCount < 0) {
+        logging.error(this, s"invalid activation count after warming < 1 ${newData}")
```

Comment:
Again fail fast via assertion as it would be a logical failure

Reply:
same as ☝️